cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-westend/src/tests/transact.rs

```diff
@@ -44,7 +44,7 @@ fn transfer_and_transact_in_same_xcm(
 	let asset_hub_location = BridgeHubWestend::sibling_location_of(AssetHubWestend::para_id());
 
 	// TODO(https://github.com/paritytech/polkadot-sdk/issues/6197): dry-run to get local fees, for now use hardcoded value.
-	let ah_fees_amount = 90_000_000_000u128; // current exact value 79_948_099_299
+	let ah_fees_amount = 700_000_000_000u128; // TODO not sure why this increased so much
```

Comment:
Hmm, I don't like the uncertainty with this `TODO`. We should try understand what is happening here.

Was this test not failing in https://github.com/paritytech/polkadot-sdk/pull/7402?

Reply:
Extracting the Weth to a common place had some side effects. Some tests used a different weth address, with an asset owner that was not Snowbridge sovereign. Using the same Weth address (that is registered in the [emulated tests genesis config](https://github.com/paritytech/polkadot-sdk/blob/master/cumulus/parachains/integration-tests/emulated/chains/parachains/assets/asset-hub-westend/src/genesis.rs#L120)), resulted in having to make some changes to the test setup and asset owner. The fee increase doesn't make sense though, will investigate.

Reply:
This was an interesting one. The root cause is the `min_balance` of the registered Weth asset. In the original version of this test, the `min_balance` of Weth was `1_000`. In the updated test, the `min_balance` was `1_000_000_000`. The weight purchased should be at least the minimum balance of the asset provided:
- https://github.com/paritytech/polkadot-sdk/blob/master/cumulus/primitives/utility/src/lib.rs#L178-L182
- https://github.com/paritytech/polkadot-sdk/blob/master/cumulus/parachains/common/src/xcm_config.rs#L61
- https://github.com/paritytech/polkadot-sdk/blob/master/substrate/frame/assets/src/types.rs#L346-L360

I set the `min_balance` back to `1_000`, which lowered the required weight fee again.

Reply:
We need to think about the impact on mainnet for using Ether and Weth to pay fees. Both their `min_balance` are `15_000_000_000_000`. Maybe this is not a problem, just thinking about it explicitly.

Reply:
I'd assume on mainnet we will create a liquidity pool (ETH, DOT) and rely on `SwapFirstAssetTrader` as the primary trader. IIUC, `TakeFirstAssetTrader` here is just a fallback without a liquidity pool, in this case, `ED` acts as a simplified version of the swap rate.

Reply:
> I'd assume on mainnet we will create a liquidity pool (ETH, DOT) and rely on SwapFirstAssetTrader as the primary trader. IIUC, TakeFirstAssetTrader here is just a fallback without a liquidity pool, in this case, ED acts as a simplified version of the swap rate.

correct - `SwapFirstAssetTrader` is tried first, `TakeFirstAssetTrader` is used as backup - `TakeFirstAssetTrader` will be removed very soon and only swaps will be used for paying fees with anything else other than DOT